### DISCUSSION TITLE
generator object instead of numpy array

### DISCUSSION THREAD

**Discussion by franegrao** [giswqs/geebook#9] [Q&A] [answered]
Hello,
pg. 95, arr = geemap.image_to_numpy(in_cog) 
is producing a generator object instead of a numpy array:
![image](https://github.com/giswqs/geebook/assets/44303502/d0ef311e-e9e1-49a0-b6a2-f48641c94558)
This is in my environment where I manually added gdal, rasterio, and localtileserver. In the environment with only pygis I get an error message that these modules are missing.
I am starting to wonder if it would make sense to try out the installation on linux or maybe wait until bugs are fixed in pygis? Any opinion?

**Comment by franegrao** [comment]
Hello again. 
I've had run into other issues before this that I didn't report and also am running into new issues on trying to proceed from here, like fiona is missing when attempting to clip by mask in following section.
I thought that maybe installing a previous version of pygis might do the trick. I found no instructions on how to do this, so I tried with a new environment specifying an older version of python (3.10) and let conda do the work of finding the correct version of pygis. This solved a lot of the issues I was having with the raster display, but not all (... e.g. figure 4.6 is not reproduced correctly: local cog is not rendered at all, and remote cog is displayed in black and white and does not render in the gist_earth palette). Also, I keep getting a generator instead of an array with geemap.image_to_numpy(in_cog).
Could you share the version with which the examples in the book were run, so I try that one?

**Comment by giswqs** [comment, marked as answer]
The `image_to_numpy` bug has been fixed in https://github.com/gee-community/geemap/pull/1895. Please update the package using `geemap.update_package()` and restart the kernel to take effect. 

geemap v0.25.0 was the version I used when writting the book. Newer version might have introduced changes that break the code in the book. Feel free to report issues here.
